transkribus_fixer/cli.py

```diff
@@ -1,26 +1,43 @@
-from click import command, Choice, argument, option
+from pkg_resources import resource_filename
+from click import command, Choice, File, argument, option
 from lxml import etree as ET
 
 from .fixer import TranskribusFixer, NS
 
-@command()
-@option('-f', '--fixes', help="Fixes to apply. Repeatable", type=Choice(['reading_order', 'table', 'metadata', 'namespace']), multiple=True)
-@argument('input-file', nargs=1)
-@argument('output-file', nargs=1)
-def cli(fixes, input_file, output_file):
-    """
-    Apply the fixes to INPUT_FILE.
+FIXERS = [func[4:] for func in dir(TranskribusFixer)
+          if callable(getattr(TranskribusFixer, func)) and func.startswith('fix_')]
+FIXERDOCS = [func + ': ' + getattr(TranskribusFixer, 'fix_' + func).__doc__ for func in FIXERS]
+FIXERS.append('namespace')
+FIXERDOCS.append('namespace: Also convert PAGE namespace version from 2013 to 2019.')
+
 
-    Also converts PAGE2013 to PAGE2019 when writing to OUTPUT_FILE.
+@command(context_settings={'help_option_names': ['-h', '--help']})
+@option('-f', '--fixes', help="Fixes to apply. Repeatable [default: all].\n\n" + "\n\n".join(FIXERDOCS),
+        default=FIXERS, type=Choice(FIXERS), multiple=True)
+@option('-I', '--prefer-imgurl', help="use TranskribusMetadata/@imgUrl for @imageFilename if available", is_flag=True)
+@option('-V', '--validate', help="Validate output against schema.", is_flag=True)
+@argument('input-file', type=File('r'), nargs=1)
+@argument('output-file', default='-', type=File('w'), nargs=1)
```

Comment:
Cool that this works. I would have thought you'd need to [set `allow_dash`](https://click.palletsprojects.com/en/8.0.x/api/?highlight=allow_dash) in click

Reply:
That's only needed for `click.Path` params. For `click.File` it says:
> The special value - indicates stdin or stdout depending on the mode.

Reply:
Good to know.